x-pack/solutions/security/plugins/security_solution/public/data_view_manager/components/data_view_picker/index.tsx

```diff
@@ -49,24 +51,43 @@ export const DataViewPicker = memo(({ scope, onClosePopover, disabled }: DataVie
 
   const { dataViewSpec, status } = useDataViewSpec(scope);
 
+  const isDefaultSourcerer = scope === DataViewManagerScopeName.default;
+  const updateUrlParam = useUpdateUrlParam<SourcererUrlState>(URL_PARAM_KEY.sourcerer);
+
   const dataViewId = dataViewSpec?.id;
 
+  // NOTE: this function is called in response to user interaction with the picker,
+  // hence - it is the only place where we should update the url param for the data view selection.
+  const handleChangeDataView = useCallback(
+    (id: string, indexPattern: string = '') => {
+      selectDataView({ id, scope });
+
+      if (isDefaultSourcerer) {
```

Comment:
Not for this PR, but `defaultSourcerer` is vague, but I'm not sure what to rename it to or how to classify the other scopes of timeline,analyzer, and the detections page 🤔 . Anyways, just wanted to bring it up for discussion separately